yt_dlp/extractor/masters.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import unicode_literals
+from .common import InfoExtractor
+from ..utils import (
+    unified_strdate,
+    traverse_obj,
+)
+
+
+class MastersIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?masters\.com/en_US/watch/(?P<date>\d{4}-\d{2}-\d{2})/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://www.masters.com/en_US/watch/2022-04-07/16493755593805191/sungjae_im_thursday_interview_2022.html',
+        'info_dict': {
+            'id': '16493755593805191',
+            'ext': 'mp4',
+            'title': 'Sungjae Im: Thursday Interview 2022',
+            'upload_date': '20220407',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        }
+    }]
+
+    _CONTENT_API_URL = "https://www.masters.com/relatedcontent/rest/v2/masters_v1/en/content/masters_v1_{video_id}_en"
+
+    def _real_extract(self, url):
+        video_id, upload_date = self._match_id(url), self._match_valid_url(url).group('date')
+        content_resp = self._download_json(
+            f'https://www.masters.com/relatedcontent/rest/v2/masters_v1/en/content/masters_v1_{video_id}_en',
+            video_id)
+        formats = self._extract_m3u8_formats(traverse_obj(content_resp, ('media', 'm3u8')), video_id, 'mp4')
```

Comment:
_extract_m3u8_formats_and_subtitles

Reply:
There doesn't seem to be any subs info coming from response so left it like that on seeing other extractors. Should I return None?

Reply:
it is recommended to always use _and_subs varinat of the functions for newer extractors
```suggestion
        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
            traverse_obj(content_resp, ('media', 'm3u8')), video_id, 'mp4')
```
the return `subtitles` too. It's fine if it is not present

Reply:
Done but had to revert back to old convension for **thumbnails** (getting **too many values to unpack** error)